tests/subsys/bluetooth/mesh/sensor_subsys/src/main.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <math.h>
+#include <stdint.h>
+
+#include <ztest.h>
+#include <bluetooth/mesh/properties.h>
+#include <bluetooth/mesh/sensor.h>
+#include <bluetooth/mesh/sensor_types.h>
+#include <sensor.h> // private header from the source folder
+
+/****************** mock section **********************************/
+/****************** mock section **********************************/
+
+/****************** callback section ******************************/
+/****************** callback section ******************************/
+static int raw_scalar_value_get(int r, int m, int d, int b)
+{
+	return r / (m * pow(10.0, d) * pow(2.0, b));
+}
+
+static void test_motion_sensor(void)
+{
+	const struct bt_mesh_sensor_type *sensor_type;
+	struct sensor_value in_value;
+	struct sensor_value out_value;
+	int err;
+	int test_vector[] = {0, 25, 50, 75, 100, 0xFF};
+
+	sensor_type = bt_mesh_sensor_type_get(BT_MESH_PROP_ID_MOTION_SENSED);
```

Comment:
Arguably, this might be a stabler test if this defines its own sensor type instead. This can't utilize the macros though, so up to you.

Reply:
It might also be worth considering testing more permutations of the type flags, like "has undefined" and similar

Reply:
This is scaffolding for sensor testing. I assumed to implement similar tests for every supported sensor type (as JIRA subtasks within the current task). It is possible to add a test for an arbitrary sensor with all configuration settings as well.

I mean this PR is not finishing the task just beginning :)